src/main/java/com/taskadapter/redmineapi/internal/URIConfigurator.java

```diff
@@ -96,15 +92,16 @@ public URI createURI(String query, NameValuePair... param) {
      * @return URI with auth parameter "key" if not in "basic auth mode.
      */
     private URI createURI(String query,
-                          Collection<? extends NameValuePair> origParams) {
-        final List<NameValuePair> params = new ArrayList<>(
-                origParams);
-        if (apiAccessKey != null) {
-            params.add(new BasicNameValuePair("key", apiAccessKey));
-        }
+                          Collection<? extends RequestParam> origParams) {
+
+//        final List<NameValuePair> params = new ArrayList<>(origParams);
+//        if (apiAccessKey != null) {
+//            params.add(new BasicNameValuePair("key", apiAccessKey));
+//        }
+        Set<RequestParam> requestParamSet = new HashSet<RequestParam>(origParams);
```

Comment:
if this set creation is the main reason why RequestParam instances are  considered equal if their names are equal, it is better to keep the corresponding logic right here